six/modules/c++/six.sicd/source/Utilities.cpp

```diff
@@ -489,6 +489,24 @@ void Utilities::getWidebandData(
             offset, extent, buffer);
 }
 
+Vector3 Utilities::getGroundPlaneNormal(const ComplexData& data)
+{
+    Vector3 groundPlaneNormal;
+
+    if (data.radarCollection->area.get() &&
+        data.radarCollection->area->plane.get())
+    {
+        AreaPlane areaPlane = *data.radarCollection->area->plane;
+        groundPlaneNormal = cross(areaPlane.xDirection->unitVector,
```

Comment:
This does compile and call the correct function. Should I include the namespace stuff anyway for clarity?

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Saturday, May 07, 2016 8:05 AM
To: ngageoint/six-library <six-library@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>; Author <author@noreply.github.com>
Subject: Re: [ngageoint/six-library] Add getGroundPlaneNormal() to SICD Utilities (#70)


In six/modules/c++/six.sicd/source/Utilities.cpp<http://cp.mcafee.com/d/2DRPoO820AcCQm74T4QT7D3tPrb38UsqenTPrUVBxB4QsCzASztWXPxEVKqenzqabaarVEVsovdydorcDIel_ok1iG6JQVsSGxHtend-wqek44NP_nV5Zx-XDnKnjppvVdYsyUZR4kRHFGTj7sG7DR8OJMddECQPt-hsd7bOabatTztPpesRG9px6kOrp3BYwhQzZpgDB91Vs-h2eNboISuvwWU68iFpKB9lwaNsjohv79K6UlzWNVo54pj9JAendLTppdwLQzh0qmU4t8_mk9VigunfAgzIiNFtd40wwwq82fP2kqJMg2N-MY6lyHFVEEd59EwDkQg30q91NEw2sV7ODbW6y0Qk0oeKqajoVUSS0QG4r8mLXY>:

> @@ -489,6 +489,24 @@ void Utilities::getWidebandData(

>              offset, extent, buffer);

>  }

>

> +Vector3 Utilities::getGroundPlaneNormal(const ComplexData& data)

> +{

> +    Vector3 groundPlaneNormal;

> +

> +    if (data.radarCollection->area.get() &&

> +        data.radarCollection->area->plane.get())

> +    {

> +        AreaPlane areaPlane = *data.radarCollection->area->plane;

> +        groundPlaneNormal = cross(areaPlane.xDirection->unitVector,

Does this compile? Don't you need a math::linear:: in front of this?

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<http://cp.mcafee.com/d/5fHCN0q43qb3yryqrzPxKVJBxAsed7bXVJYsOMOyqejhOrhKZtVMQsTd7bNJ55B5dYQsKcfCN6IdCjS7a_Ia0Fl3mWsKrlgRKDbC_gd7a22oV_HYy-M_tPHTbFIILYC-ehsuWyaqRQRrFzKl3PWApmU6CQPqpK_8K6zBV55BeXNKVIDeqR4IMzapdIxO-g8Wh-IEjOAwYKv8x7oBImrffMtv7p7_j6buMgsguTn1PdQ467H6zBV511RxN13mmn4TbEzA7wcyUCMy-ejsdMH7RzOMa8OCjr8sKrvKOOr1vF6y0QJM8Wh-IEjOAwYKv8x7oBziWq81110Qg4vC4ERrww5zZxUcH5njPhgqajh1eFEw60Qi3zh04VOfBenQd41EE0MtsQkCNPNKmlQe1YF>


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.